src/main/java/fi/okm/jod/yksilo/config/SecurityConfig.java

```diff
@@ -47,6 +49,18 @@
 @Slf4j
 public class SecurityConfig {
 
+  @Value("${apiKey}")
+  private String apiKey;
+
+  @Bean
+  @Order(2)
+  public SecurityFilterChain externalApiFilterChain(HttpSecurity http) throws Exception {
+    http.securityMatcher("/external-api/**")
+        .addFilterBefore(new ApiKeyFilter(this.apiKey), UsernamePasswordAuthenticationFilter.class)
+        .authorizeHttpRequests(auth -> auth.anyRequest().authenticated());
+    return http.build();
+  }
+
```

Comment:
It would be good to refactor this to separate configuration object co-located with other external api classes.